pyfatfs/PyFat.py

```diff
@@ -1019,3 +990,230 @@ def open_fs(filename: str, offset: int = 0,
         pf.open(filename)
         yield pf
         pf.close()
+
+    def mkfs(self, filename: str, size: int = None,
+             fat_type: Union["PyFat.FAT_TYPE_FAT12",
+                             "PyFat.FAT_TYPE_FAT16",
+                             "PyFat.FAT_TYPE_FAT32"] = None,
+             sector_size: int = 512,
+             number_of_fats: int = 2,
+             label: str = "NO NAME",
+             media_type: int = 0xF8):
+        """Create a new FAT filesystem.
+
+        :param filename: `str`: Name of file to create filesystem in
+        :param size: `int`: Size of new filesystem in bytes
+        :param fat_type: `FAT_TYPE_FAT{12,16,32}`: FAT type
+        :param sector_size: `int`: Size of a sector in bytes
+        :param number_of_fats: `int`: Number of FATs on the disk
+        :param label: `str`: Volume label
+        :param media_type: `int`: Media type (0xF{0,8-F})
+        """
+        self.initialized = True
+        self.is_read_only = False
+        self.fat_type = fat_type
+        self.__set_fp(open(filename, mode='rb+'))
+
+        if size is None:
+            size = self.__fp_offset - self.__fp.seek(-1)
+
+        self.__fp.truncate(size)
```

Comment:
This will fail if we have a fixed-size file (i.e., block device) on overprovisioning. Should be handled.